control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -426,9 +425,9 @@ func (r *Reconciler) finalizeKind(ctx context.Context, channel *messagingv1beta1
 	// 	See (under discussions KIPs, unlikely to be accepted as they are):
 	// 	- https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=181306446
 	// 	- https://cwiki.apache.org/confluence/display/KAFKA/KIP-286%3A+producer.send%28%29+should+not+block+on+metadata+update
-	address := receiver.Address(r.IngressHost, channel)
-	proberAddressable := prober.Addressable{
-		Address: address,
+
+	proberAddressable := prober.NewAddressable{
+		AddressStatus: &duckv1.AddressStatus{},
```

Comment:
@Rahul-Kumar-prog is there a reason why the `AddressStatus` is empty here? It looks like there was an actual address being passed in before, not just an empty address

Reply:
If you want this change to be backwards compatible, you will need to set `AddressStatus: &duckv1.AddressStatus{Addresses: []duckv1.Addressable{address}}

Reply:
The way the composite prober works is it will iterate through all the addresses in AddressStatus.Addresses and probe each of them